.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,131 @@
+---
+name: Build and Publish
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build econet-${{ matrix.appliance.shorthand }}-${{ matrix.platform }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [esp32, esp8266]
+        appliance:
+          - name: electric_tank_water_heater
+            shorthand: etwh
+          - name: heatpump_water_heater
+            shorthand: hpwh
+          - name: hvac
+            shorthand: hvac
+          - name: tankless_water_heater
+            shorthand: tlwh
+      fail-fast: false
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4.1.1
+      - name: Build manifest
+        uses: mikefarah/yq@v4.35.2
+        with:
+          cmd: yq -P '.packages.econet = "econet_${{ matrix.appliance.name }}.yaml" | .packages.econet   tag = "!include"' ./build-yaml/${{ matrix.platform }}.yaml > econet-${{ matrix.appliance.shorthand }}-${{ matrix.platform }}.yaml
+      - name: Build firmware
+        uses: esphome/build-action@v1.8.0
+        id: esphome-build
+        with:
+          yaml_file: econet-${{ matrix.appliance.shorthand }}-${{ matrix.platform }}.yaml
+          version: latest
+      - name: Copy firmware and manifest
+        run: |
+          mkdir output
+          mv ${{ steps.esphome-build.outputs.name }} output/
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3.1.3
+        with:
+          name: econet-${{ matrix.appliance.shorthand }}-${{ matrix.platform }}
+          path: output
+          retention-days: 7
+
+  consolidate-manifests:
+    # if: github.event_name != 'pull_request' && github.ref == 'refs/heads/main'
+    name: Consolidate ${{ matrix.appliance.shorthand }} manifests
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      matrix:
+        appliance:
```

Comment:
Is it possible to define this map once?

Reply:
I've been trying to figure out a way but haven't managed to. Perhaps there is some native YAML way to include/repeat a section but nothing I've been able to find for that or supported by Github actions for capturing "snippets" like this.